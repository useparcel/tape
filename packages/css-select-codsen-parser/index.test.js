import parse from "codsen-parser";
import { selectAll, selectOne } from "css-select";
import { prepare, adapter } from "./index.js";

describe("css-select-codsen-parser", () => {
  test("integration test", () => {
    const ast = prepare(
      parse(`
        <div id="greeting">
          Hello <span class="name" data-attribute>Alice</span>
          Hello <span class="name">Bob</span>
        </div>
      `)
    );

    // queries attribute existence, attribute value, sibling, text,
    const query = "#greeting span[data-attribute] + span.name:icontains(bob)";
    const nodes = selectAll(query, ast, { adapter });
    expect(nodes).toMatchSnapshot();

    const node = selectOne(query, ast, { adapter });
    expect(node).toMatchSnapshot();
  });
});
