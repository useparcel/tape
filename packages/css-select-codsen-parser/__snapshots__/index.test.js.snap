// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`css-select-codsen-parser integration test 1`] = `
Array [
  Object {
    "attribs": Array [
      Object {
        "attribClosingQuoteAt": 125,
        "attribEnds": 126,
        "attribLeft": 112,
        "attribName": "class",
        "attribNameEndsAt": 119,
        "attribNameRecognised": true,
        "attribNameStartsAt": 114,
        "attribOpeningQuoteAt": 120,
        "attribStarts": 114,
        "attribValue": Array [
          Object {
            "end": 125,
            "start": 121,
            "type": "text",
            "value": "name",
          },
        ],
        "attribValueEndsAt": 125,
        "attribValueRaw": "name",
        "attribValueStartsAt": 121,
      },
    ],
    "children": Array [
      Object {
        "end": 130,
        "parent": [Circular],
        "start": 127,
        "type": "text",
        "value": "Bob",
      },
    ],
    "closing": false,
    "end": 127,
    "kind": "inline",
    "parent": Object {
      "attribs": Array [
        Object {
          "attribClosingQuoteAt": 26,
          "attribEnds": 27,
          "attribLeft": 12,
          "attribName": "id",
          "attribNameEndsAt": 16,
          "attribNameRecognised": true,
          "attribNameStartsAt": 14,
          "attribOpeningQuoteAt": 17,
          "attribStarts": 14,
          "attribValue": Array [
            Object {
              "end": 26,
              "start": 18,
              "type": "text",
              "value": "greeting",
            },
          ],
          "attribValueEndsAt": 26,
          "attribValueRaw": "greeting",
          "attribValueStartsAt": 18,
        },
      ],
      "children": Array [
        Object {
          "end": 45,
          "parent": [Circular],
          "start": 28,
          "type": "text",
          "value": "
          Hello ",
        },
        Object {
          "attribs": Array [
            Object {
              "attribClosingQuoteAt": 62,
              "attribEnds": 63,
              "attribLeft": 49,
              "attribName": "class",
              "attribNameEndsAt": 56,
              "attribNameRecognised": true,
              "attribNameStartsAt": 51,
              "attribOpeningQuoteAt": 57,
              "attribStarts": 51,
              "attribValue": Array [
                Object {
                  "end": 62,
                  "start": 58,
                  "type": "text",
                  "value": "name",
                },
              ],
              "attribValueEndsAt": 62,
              "attribValueRaw": "name",
              "attribValueStartsAt": 58,
            },
            Object {
              "attribClosingQuoteAt": null,
              "attribEnds": 78,
              "attribLeft": 62,
              "attribName": "data-attribute",
              "attribNameEndsAt": 78,
              "attribNameRecognised": false,
              "attribNameStartsAt": 64,
              "attribOpeningQuoteAt": null,
              "attribStarts": 64,
              "attribValue": Array [],
              "attribValueEndsAt": null,
              "attribValueRaw": null,
              "attribValueStartsAt": null,
            },
          ],
          "children": Array [
            Object {
              "end": 84,
              "parent": [Circular],
              "start": 79,
              "type": "text",
              "value": "Alice",
            },
          ],
          "closing": false,
          "end": 79,
          "kind": "inline",
          "parent": [Circular],
          "pureHTML": true,
          "recognised": true,
          "start": 45,
          "tagName": "span",
          "tagNameEndsAt": 50,
          "tagNameStartsAt": 46,
          "type": "tag",
          "value": "<span class=\\"name\\" data-attribute>",
          "void": false,
        },
        Object {
          "attribs": Array [],
          "children": Array [],
          "closing": true,
          "end": 91,
          "kind": "inline",
          "parent": [Circular],
          "pureHTML": true,
          "recognised": true,
          "start": 84,
          "tagName": "span",
          "tagNameEndsAt": 90,
          "tagNameStartsAt": 86,
          "type": "tag",
          "value": "</span>",
          "void": false,
        },
        Object {
          "end": 108,
          "parent": [Circular],
          "start": 91,
          "type": "text",
          "value": "
          Hello ",
        },
        [Circular],
        Object {
          "attribs": Array [],
          "children": Array [],
          "closing": true,
          "end": 137,
          "kind": "inline",
          "parent": [Circular],
          "pureHTML": true,
          "recognised": true,
          "start": 130,
          "tagName": "span",
          "tagNameEndsAt": 136,
          "tagNameStartsAt": 132,
          "type": "tag",
          "value": "</span>",
          "void": false,
        },
        Object {
          "end": 146,
          "parent": [Circular],
          "start": 137,
          "type": "text",
          "value": "
        ",
        },
      ],
      "closing": false,
      "end": 28,
      "kind": null,
      "parent": null,
      "pureHTML": true,
      "recognised": true,
      "start": 9,
      "tagName": "div",
      "tagNameEndsAt": 13,
      "tagNameStartsAt": 10,
      "type": "tag",
      "value": "<div id=\\"greeting\\">",
      "void": false,
    },
    "pureHTML": true,
    "recognised": true,
    "start": 108,
    "tagName": "span",
    "tagNameEndsAt": 113,
    "tagNameStartsAt": 109,
    "type": "tag",
    "value": "<span class=\\"name\\">",
    "void": false,
  },
]
`;

exports[`css-select-codsen-parser integration test 2`] = `
Object {
  "attribs": Array [
    Object {
      "attribClosingQuoteAt": 125,
      "attribEnds": 126,
      "attribLeft": 112,
      "attribName": "class",
      "attribNameEndsAt": 119,
      "attribNameRecognised": true,
      "attribNameStartsAt": 114,
      "attribOpeningQuoteAt": 120,
      "attribStarts": 114,
      "attribValue": Array [
        Object {
          "end": 125,
          "start": 121,
          "type": "text",
          "value": "name",
        },
      ],
      "attribValueEndsAt": 125,
      "attribValueRaw": "name",
      "attribValueStartsAt": 121,
    },
  ],
  "children": Array [
    Object {
      "end": 130,
      "parent": [Circular],
      "start": 127,
      "type": "text",
      "value": "Bob",
    },
  ],
  "closing": false,
  "end": 127,
  "kind": "inline",
  "parent": Object {
    "attribs": Array [
      Object {
        "attribClosingQuoteAt": 26,
        "attribEnds": 27,
        "attribLeft": 12,
        "attribName": "id",
        "attribNameEndsAt": 16,
        "attribNameRecognised": true,
        "attribNameStartsAt": 14,
        "attribOpeningQuoteAt": 17,
        "attribStarts": 14,
        "attribValue": Array [
          Object {
            "end": 26,
            "start": 18,
            "type": "text",
            "value": "greeting",
          },
        ],
        "attribValueEndsAt": 26,
        "attribValueRaw": "greeting",
        "attribValueStartsAt": 18,
      },
    ],
    "children": Array [
      Object {
        "end": 45,
        "parent": [Circular],
        "start": 28,
        "type": "text",
        "value": "
          Hello ",
      },
      Object {
        "attribs": Array [
          Object {
            "attribClosingQuoteAt": 62,
            "attribEnds": 63,
            "attribLeft": 49,
            "attribName": "class",
            "attribNameEndsAt": 56,
            "attribNameRecognised": true,
            "attribNameStartsAt": 51,
            "attribOpeningQuoteAt": 57,
            "attribStarts": 51,
            "attribValue": Array [
              Object {
                "end": 62,
                "start": 58,
                "type": "text",
                "value": "name",
              },
            ],
            "attribValueEndsAt": 62,
            "attribValueRaw": "name",
            "attribValueStartsAt": 58,
          },
          Object {
            "attribClosingQuoteAt": null,
            "attribEnds": 78,
            "attribLeft": 62,
            "attribName": "data-attribute",
            "attribNameEndsAt": 78,
            "attribNameRecognised": false,
            "attribNameStartsAt": 64,
            "attribOpeningQuoteAt": null,
            "attribStarts": 64,
            "attribValue": Array [],
            "attribValueEndsAt": null,
            "attribValueRaw": null,
            "attribValueStartsAt": null,
          },
        ],
        "children": Array [
          Object {
            "end": 84,
            "parent": [Circular],
            "start": 79,
            "type": "text",
            "value": "Alice",
          },
        ],
        "closing": false,
        "end": 79,
        "kind": "inline",
        "parent": [Circular],
        "pureHTML": true,
        "recognised": true,
        "start": 45,
        "tagName": "span",
        "tagNameEndsAt": 50,
        "tagNameStartsAt": 46,
        "type": "tag",
        "value": "<span class=\\"name\\" data-attribute>",
        "void": false,
      },
      Object {
        "attribs": Array [],
        "children": Array [],
        "closing": true,
        "end": 91,
        "kind": "inline",
        "parent": [Circular],
        "pureHTML": true,
        "recognised": true,
        "start": 84,
        "tagName": "span",
        "tagNameEndsAt": 90,
        "tagNameStartsAt": 86,
        "type": "tag",
        "value": "</span>",
        "void": false,
      },
      Object {
        "end": 108,
        "parent": [Circular],
        "start": 91,
        "type": "text",
        "value": "
          Hello ",
      },
      [Circular],
      Object {
        "attribs": Array [],
        "children": Array [],
        "closing": true,
        "end": 137,
        "kind": "inline",
        "parent": [Circular],
        "pureHTML": true,
        "recognised": true,
        "start": 130,
        "tagName": "span",
        "tagNameEndsAt": 136,
        "tagNameStartsAt": 132,
        "type": "tag",
        "value": "</span>",
        "void": false,
      },
      Object {
        "end": 146,
        "parent": [Circular],
        "start": 137,
        "type": "text",
        "value": "
        ",
      },
    ],
    "closing": false,
    "end": 28,
    "kind": null,
    "parent": null,
    "pureHTML": true,
    "recognised": true,
    "start": 9,
    "tagName": "div",
    "tagNameEndsAt": 13,
    "tagNameStartsAt": 10,
    "type": "tag",
    "value": "<div id=\\"greeting\\">",
    "void": false,
  },
  "pureHTML": true,
  "recognised": true,
  "start": 108,
  "tagName": "span",
  "tagNameEndsAt": 113,
  "tagNameStartsAt": 109,
  "type": "tag",
  "value": "<span class=\\"name\\">",
  "void": false,
}
`;
